lib/uksignal/signal.c

```diff
@@ -297,20 +339,45 @@ int sigwait(const sigset_t *set, int *sig)
 	ptr->wait.status = UK_SIG_NOT_WAITING;
 
 	/* do not execute handler, set received signal */
-	*sig = ptr->wait.received_signal.si_signo;
+	*info = ptr->wait.received_signal;
 
 	/* execute other pending signals */
 	uk_sig_handle_signals();
 
 	return 0; /* returns positive errno */
 }
 
+#if UK_LIBC_SYSCALLS
+int sigwait(const sigset_t *set, int *sig)
+{
+	int error;
+	siginfo_t si;
+
+	error = rt_sigtimedwait(set, &si, NULL, (_NSIG / 8));
+	*sig = si.si_signo;
+
+	return error;
+}
+#endif
+
+UK_SYSCALL_R_DEFINE(int, tkill, pid_t __unused, pid, int, sig)
+{
+	return uk_sig_thread_kill(uk_thread_current(), sig);
+}
+
+#if UK_LIBC_SYSCALLS
+int raise(int sig)
+{
+	return tkill(-1, sig);
+}
+#endif
+
 /*
  * Search for a thread that does not have the signal blocked
  * If all of the threads have the signal blocked, add it to process
  * pending signals
  */
-int kill(pid_t pid, int sig)
+UK_SYSCALL_R_DEFINE(int, kill, pid_t, pid, int, sig)
```

Comment:
This function still sets `errno` in lines 398 and 403. This should be:
```C
	if (pid != 1 && pid != 0 && pid != -1)
		return -ESRCH;

	if (!uk_sig_is_valid(sig))
		return -EINVAL;
```

You can also remove the extra empty line 396.